src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Expressions/Parsers/SearchParameterExpressionParser.cs

```diff
@@ -93,7 +93,7 @@ public class SearchParameterExpressionParser : ISearchParameterExpressionParser
                         string.Format(CultureInfo.InvariantCulture, Core.Resources.ModifierNotSupported, modifier, searchParameter.Name));
                 }
 
-                outputExpression = Expression.Contains(FieldName.TokenText, null, value, true);
+                outputExpression = Expression.StartsWith(FieldName.TokenText, null, value, true);
```

Comment:
I believe for tokens, the ':text" modifier is actually a contains. 

From the docs (https://www.hl7.org/fhir/search.html): 
GET [base]/Condition?code:text=headache -> Search for any Condition with a code that has a text "headache" associated with it (either in the text, or a display)

Reply:
From the spec:
>`:text` [...] In this case, the search functions as a normal string search.

[...]

>This search is insensitive to casing and included combining characters, like accents or other diacritical marks. Punctuation and non-significant whitespace (e.g. repeated space characters, tab vs space) should also be ignored. By default, a field matches a string query if the value of the field equals or starts with the supplied parameter value, after both have been normalized by case and combining characters. 

Also hapi does not implement it as a contains.
http://hapi.fhir.org/baseDstu3/Observation?code:text=Respiratory matches "Respiratory Rate"
http://hapi.fhir.org/baseDstu3/Observation?code:text=Respiratory&code:text=Rate does not

Neither does Vonk:
https://vonk.fire.ly/Observation?_format=json&code:text=Respiratory&code:text=rate